### PR TITLE
165 AooEoo failed without notification

### PR DESCRIPTION
**Pull request by qifeng-bai**
Handle error messages return from Tasks
1, Error message may come with Status error code, may with 200
2, Error message may be a plain text
3, Error message may be JSON

### CODE REVIEW

File: grails-app/assets/javascripts/spApp/service/toolsService.js [hunk 1]
```diff
@@ -1,430 +1,470 @@
...
-                        // Create log entry for an external tool when the task is created with the taskId
-                        LoggerService.log('Tool', uiScope.toolName, '{ "taskId": "' + uiScope.externalTaskId + '"}')
-
-                        $timeout(function () {
-                            _checkStatus(uiScope);
-                        }, 5000)
-                    });
-
-                    return $q.when(false)
+  'use strict';
```
Comment (nickdos) [inline grails-app/assets/javascripts/spApp/service/toolsService.js:2]: You changed file from 4 space indent to 2 space. ALA coding style is 4 space. Pls fix.

File: grails-app/assets/javascripts/spApp/service/toolsService.js [hunk 2]
```diff
@@ -1,430 +1,470 @@
...
+                  LoggerService.log('Tool', uiScope.toolName,
+                      '{ "taskId": "' + uiScope.externalTaskId + '"}')
+
+                  $timeout(function () {
+                    _checkStatus(uiScope);
+                  }, 5000)
+                },
+                function (error) {
+                    var message = '';
+                  if(error.headers()['content-type'].indexOf('json')>0){
```
Comment (nickdos) [inline grails-app/assets/javascripts/spApp/service/toolsService.js:60]: indentation is wrong (`if` is to left of line above it) - did you mix spaces and tabs? Use IDE to format the code.

**Review by nickdos — COMMENTED** [review]
Change of indentation resulted in many more lines of code changed than needed. Also was not ALA code style format - 4 spaces. Needs fixing

File: grails-app/assets/javascripts/spApp/service/toolsService.js (hunk 2 above)
Comment (qifeng-bai) [reply grails-app/assets/javascripts/spApp/service/toolsService.js:60]: Reverted and add codes in

File: grails-app/assets/javascripts/spApp/service/toolsService.js (hunk 1 above)
Comment (qifeng-bai) [reply grails-app/assets/javascripts/spApp/service/toolsService.js:2]: Fixed

**Review by nickdos — APPROVED** [review]

**Pull request closed without merging**